nibabel/filebasedimages.py

```diff
@@ -47,19 +52,19 @@ def from_header(klass, header=None):
         )
 
     @classmethod
-    def from_fileobj(klass, fileobj: io.IOBase):
-        raise NotImplementedError
+    def from_fileobj(klass: type[HdrT], fileobj: io.IOBase) -> HdrT:
+        raise NotImplementedError  # pragma: no cover
```

Comment:
From the SO link you sent re the `object` vs. `Any` comment, I also learned that it is better to:
```suggestion
        return NotImplementedError  # pragma: no cover
```
[ref](https://stackoverflow.com/questions/878943/why-return-notimplemented-instead-of-raising-notimplementederror)